### PR TITLE
Bugfix operationbased with interface not generated interface model as sealed interface

### PR DESCRIPTION
**Pull request by chalermpong**
Fix for #4421

### CODE REVIEW

**Comment by chalermpong** [comment]
There is an unwanted side effect on responseBased codegen. Need to revise the solution.

**Comment by chalermpong** [comment]
I revised the solution. Ready for review.

File: libraries/apollo-compiler/src/test/graphql/com/example/fragment_with_inline_fragment/kotlin/operationBased2/fragment_with_inline_fragment/TestQuery.kt.expected [hunk 1]
```diff
@@ -1,124 +0,0 @@
-//
-// AUTO-GENERATED FILE. DO NOT MODIFY.
-//
-// This class was automatically generated by Apollo GraphQL version '$VERSION'.
-//
-package com.example.fragment_with_inline_fragment
```
Comment (martinbonnin) [inline libraries/apollo-compiler/src/test/graphql/com/example/fragment_with_inline_fragment/kotlin/operationBased2/fragment_with_inline_fragment/TestQuery.kt.expected:6]: Thanks for removing these 🙏

**Review by martinbonnin — APPROVED** [review]
LGTM, thanks for fixing this!

@BoD looking at this, I believe we're missing a few `sealed` interfaces for `responseBased` fragment implementations. Any reason we do not pass `true` for [`FragmentBuilder`](https://github.com/apollographql/apollo-kotlin/blob/05f2a3295517fb7a64c2259c38b453a37f3c4d19/libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/kotlin/file/FragmentBuilder.kt#L50)?

**Comment by BoD** [comment]
> @BoD looking at this, I believe we're missing a few sealed interfaces for responseBased fragment implementations. Any reason we do not pass true for [FragmentBuilder](https://github.com/apollographql/apollo-kotlin/blob/05f2a3295517fb7a64c2259c38b453a37f3c4d19/libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/kotlin/file/FragmentBuilder.kt#L50)?

@martinbonnin No, I can't think of a reason, this was probably an oversight. The generated code looks good with `true` here!

**Comment by martinbonnin** [comment]
Alright, merging this and I'll make a follow up PR for `FragmentBuilder`

**Pull request merged**